### ISSUE TITLE
azurerm_data_factory_integration_runtime_azure_ssis catalog_info's pricing_tier doesn't support GP_S_Gen5_1

### ISSUE THREAD

**Issue by dvansteenburg** [hashicorp/terraform-provider-azurerm#14092]
I am using azurerm version 2.84.0, terraform version 1.0.4

I am using an Azure SQL Server to host my SSISDB database (not an SQL Managed Instance) for my data factory.  The options of Basic, Standard, Premium, and PremiumRS are not recognized by Data Factory when it is trying to create the SSISDB the first time the SSIS Integration starts up.  

Here is the configuration I used with azurerm_data_factory_integration_runtime_azure_ssis...

```
resource "azurerm_data_factory_integration_runtime_azure_ssis" "this" {
 data_factory_name                = "data-factory-XXXXXXX"
 description                      = "ssis-XXXXXXXX"
 edition                          = "Standard"
 license_type                     = "LicenseIncluded"
 location                         = "eastus2"
 max_parallel_executions_per_node = 1
 name                             = "ssis-XXXXXXXX"
 node_size                        = "Standard_D2_v3"
 number_of_nodes                  = 1
 resource_group_name              = "rg-XXXXXXX"

 catalog_info {
   administrator_login    = "SqlXXXXX"
   administrator_password = "PWDXXXXXX"
   pricing_tier           = "Standard"
   server_endpoint        = "dev-XXXXXXX.database.windows.net"
 }

 vnet_integration {
  subnet_name = "subnet-XXXXXXXXXXXXXXXXXXXX"
  vnet_id     = "/subscriptions/XXXXXXXX-xxxx-xxxx-xxxx-XXXXXXXXXXXX/resourceGroups/rg-XXXXXXX/providers/Microsoft.Network/virtualNetworks/vnet-XXXXXXX"
        }
    }
```
... and I get the following error when I start up the newly created SSIS Integration Runtime in Data Factory...

```
Error 1:
  Last operation 'Start' get the status 'Failed'.
  Error code: InvalidCatalogPricingTier
  Error message: 'catalogPricingTier' property is invalid. Code : 40804, Message : The specified service objective 'Standard' is invalid.
```

I tried setting the pricing_tier to "GP_S_Gen5_1", but that also did not work as terraform identified it as not a valid pricing tier, with the following error...

```
Error: expected catalog_info.0.pricing_tier to be one of [Basic Standard Premium PremiumRS], got GP_S_Gen5_1
```

**Comment by dvansteenburg** [comment]
In further research, this module incorrectly tries to verify the pricing_tier values to only 5 valid values.  However, in Data Factory, there are 3 dozen or so values which show up in the Portal, than this module needs to be modified to ignore those 5 values.  Additionally, if you try to deploy this against a SQL Managed Instance, the pricing_tier cannot be included as there are no pricing tiers for a SQL Managed Instance, so this needs to be able to either be excluded and/or accept a NULL option.

**Comment by dvansteenburg** [comment]
The documentation needs to be updated, but this change was released in v2.86.0.
